sc-multisig-specs.md

```diff
@@ -0,0 +1,80 @@
+# Multisig Smart Contract (MSC)
+
+## Abstract
+Cryptocurrencies can be one of the safest ways to store and manage wealth and value. By safeguarding a short list of words, the so-called seed or recovery phrase, anyone can protect thousands or millions of dollars in wealth and rest assured that no hacker or government can take it from them. In practice, it’s never so easy.
+
+One problem is that single-signature addresses rely on protecting a single private key.
+
+A better solution would use for example 2-of-3 multisig (or any combination of M-of-N for M ≤ N) quorum consisting of three separate private keys, held by three separate people or entities, and requiring any two to sign. This provides both security and redundancy since compromising any one key/person does not break the quorum: if one key is stolen or lost, the other two keyholders can sweep funds to another address (protected by a new quorum) by mutually signing a transaction moving the funds.
+
+As an example, let us imagine the following scenario. An institution launches a stablecoin. For safety, it is required that 3 out of 5 designated addresses sign any mint or burn transaction. Alice deploys the multisig SC. She adds Bob, Charlie, Dave and Eve as signers to the contract and sets the quorum to a minimum number of signers to 3. A quorum of signatures is also required to add or remove signers after the initial deployment. If for some reason, Eve’s account is compromised, Alice proposes removing Eve’s address from the signers’ board. Charlie and Dave sign, causing Eve’s address to be removed. There are only 4 addresses now registered in the contract. By the same process, signers could add 2 more addresses to their ranks, and increase the required quorum signatures from 3 to 4.
+
+Thus, essentially the multisig SC (we will refer to it, from now on, as MSC) enables multiple parties to sign or approve an action that takes place - typically a requirement for certain wallets, accounts, and smart contracts to prevent a rogue or hacked individual from performing detrimental actions.
+
+## Multisig transaction flow
+On-chain multisig wallets are made possible by the fact that smart contracts can call other smart contracts. To execute a multisig transaction the flow would be:
+
+* A proposer or board member proposes an action.
+* The proposed action receives a unique id/hash.
+* N board members are notified (off-chain) to review the action with the specific id/hash.
+* M out of N board members sign and approve the action.
+* Any proposer or board member “performs the action”.
+
+## Design guidelines
+ 
```

Comment:
It is not very clear how this is implemented: through a deployable SC or a system (internal) SC. I assume that this is a deployable SC written in Rust and compiled in WASM so we might state that this is a deployable SC.

Reply:
yes, it is a deployable SC written in Rust and compiled in WASM. Added this for clarity.